import grpc
import sovereign_pb2
import sovereign_pb2_grpc

def run_test():
    # Rustã‚µãƒ¼ãƒãƒ¼ã«æ¥ç¶š
    channel = grpc.insecure_channel('localhost:50051')
    stub = sovereign_pb2_grpc.ExecutorStub(channel)

    print("1ï¸âƒ£ æ™®é€šã«å–å¼•ã‚’ãƒªã‚¯ã‚¨ã‚¹ãƒˆã—ã¦ã¿ã‚‹...")
    req1 = sovereign_pb2.TradeRequest(side="BUY", price=150.0, expected_e=0.05)
    res1 = stub.ExecuteTrade(req1)
    print(f"çµæœ: {res1.message} (Success: {res1.success})\n")

    print("2ï¸âƒ£ ğŸ†˜ ç·Šæ€¥åœæ­¢ï¼ˆKill-Switchï¼‰ã‚’ç™ºå‹•ï¼")
    stop_req = sovereign_pb2.StopRequest(reason="ãƒ†ã‚¹ãƒˆï¼šç•°å¸¸æ¤œçŸ¥ã®ã‚·ãƒŸãƒ¥ãƒ¬ãƒ¼ã‚·ãƒ§ãƒ³")
    stop_res = stub.EmergencyStop(stop_req)
    print(f"çµæœ: {stop_res.message}\n")

    print("3ï¸âƒ£ åœæ­¢å¾Œã«ã‚‚ã†ä¸€åº¦å–å¼•ã‚’ãƒªã‚¯ã‚¨ã‚¹ãƒˆã—ã¦ã¿ã‚‹...")
    res2 = stub.ExecuteTrade(req1)
    print(f"çµæœ: {res2.message} (Success: {res2.success})")

    if not res2.success:
        print("\nâœ… ãƒ†ã‚¹ãƒˆæˆåŠŸï¼ã‚·ã‚¹ãƒ†ãƒ ã¯å®Œå…¨ã«ãƒ­ãƒƒã‚¯ã•ã‚Œã¦ã„ã¾ã™ã€‚")
    else:
        print("\nâŒ ãƒ†ã‚¹ãƒˆå¤±æ•—ã€‚ãƒ­ãƒƒã‚¯ãŒåŠ¹ã„ã¦ã„ã¾ã›ã‚“ã€‚")

if __name__ == "__main__":
    run_test()
